packages/zoe/test/unitTests/contracts/loan/test-loan-e2e.js

```diff
@@ -0,0 +1,108 @@
+// ts-check
+import '../../../../exported';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+import { E } from '@agoric/eventual-send';
+
+import bundleSource from '@agoric/bundle-source';
+import { makeSubscriptionKit } from '@agoric/notifier';
+
+import { checkDetails, checkPayout } from './helpers';
+import { setup } from '../../setupBasicMints';
+import { makeFakePriceAuthority } from '../../../fakePriceAuthority';
+import buildManualTimer from '../../../../tools/manualTimer';
+
+const loanRoot = `${__dirname}/../../../../src/contracts/loan/`;
+const autoswapRoot = `${__dirname}/../../../../src/contracts/autoswap`;
+
+test.todo('loan - no mmr');
+test.todo('loan - bad mmr');
+test.todo('loan - no priceAuthority');
+test.todo('loan - badPriceAuthority');
+test.todo('loan - bad autoswap, no autoswap');
+test.todo('loan - wrong keywords');
+
+test.todo('loan - lend - wrong exit rule');
+test.todo('loan - lend - must want nothing');
```

Comment:
more: 
 * when liquidity is added, there's no interference between scheduled liquidations. I'd like to see the first scheduled event being triggered first with no effect (followed by the updated trigger), as well as the updated event triggering first (and the original not interfering).
 * If the price level doesn't change, but debt accrues sufficiently, it ought to be possible to trigger liquidation. I'm not sure we schedule liquidation events aggressively enough as debt grows.

Reply:
> when liquidity is added, there's no interference between scheduled liquidations. I'd like to see the first scheduled event being triggered first with no effect (followed by the updated trigger), as well as the updated event triggering first (and the original not interfering).

I think the test 'borrow, then addCollateral, then getLiquidationPromise' covers this, but let me know if you think it doesn't.

> If the price level doesn't change, but debt accrues sufficiently, it ought to be possible to trigger liquidation. I'm not sure we schedule liquidation events aggressively enough as debt grows.

Oh, good catch! I think we have to schedule a new liquidation on every update of the debt. Let me look into this.

Reply:
> I think the test 'borrow, then addCollateral, then getLiquidationPromise' covers this, but let me know if you think it doesn't.

Let me know if I'm mis-reading the test, but it looks like no liquidation takes place. I'd like to see the case where both price triggers are hit, and the first has no effect because liquidity was increased. When the revised price trigger goes off, liquidation should take place, and we should see `t.truthy(liquidated)`.

Reply:
Liquidation is successfully triggered